File: app/src/main/java/me/zipi/navitotesla/api/TeslaApi.java
package me.zipi.navitotesla.api;

import me.zipi.navitotesla.model.ShareRequest;
import me.zipi.navitotesla.model.TeslaApiResponse;
import me.zipi.navitotesla.model.Vehicle;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface TeslaApi {

    @GET("/api/1/vehicles")
    Call<TeslaApiResponse.ListType<Vehicle>> vehicles();

    @POST("/api/1/vehicles/{id}/command/share")
    Call<TeslaApiResponse.ObjectType<TeslaApiResponse.Result>> share(@Path("id") Long id, @Body ShareRequest request);
}

File: app/src/main/kotlin/me/zipi/navitotesla/api/TeslaApi.kt
package me.zipi.navitotesla.api

import me.zipi.navitotesla.model.ShareRequest
import me.zipi.navitotesla.model.TeslaApiResponse
import me.zipi.navitotesla.model.Vehicle
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface TeslaApi {
    @GET("/api/1/vehicles")
    @Deprecated("endpoint is only available on fleetapi")
    suspend fun vehicles(): Response<TeslaApiResponse.ListType<Vehicle>>

    @GET("/api/1/products")
    suspend fun products(): Response<TeslaApiResponse.ListType<Map<String, Any>>>

    @POST("/api/1/vehicles/{id}/command/share")
    suspend fun share(
        @Path("id") id: Long,
        @Body request: ShareRequest,
    ): Response<TeslaApiResponse.ObjectType<TeslaApiResponse.Result>>
}
